File: src/main/resources/templates/java/solution/src/${packageNameFolder}/Policy.java
package ${packageName};

public class Policy {
	
	private Context context;

	public Policy(Context context) {
		this.context = context;
	}

	public void configure (){
		if(this.context.getDates().size() > 10) {
			System.out.println("More than 10 dates, choosing merge sort!");
			this.context.setSortAlgorithm(new MergeSort());
		} else {
			System.out.println("Less or equal than 10 dates. choosing quick sort!");
			this.context.setSortAlgorithm(new BubbleSort());
		}
	}
}

File: src/main/resources/templates/kotlin/solution/src/${packageNameFolder}/Policy.kt
package ${packageName}

class Policy(private var context: Context) {

    fun configure(timeIsImportant: Boolean, spaceIsImportant: Boolean) {
        if (timeIsImportant && !spaceIsImportant) {
            println("Time is important –> Merge Sort!")
            context.sortAlgorithm = MergeSort()
        } else if (timeIsImportant && spaceIsImportant) {
            println("Time & Space are important –> Quick Sort!")
            context.sortAlgorithm = QuickSort()
        }
    }
}
